### PR TITLE
Adds support for docker usage

### PR DESCRIPTION
**Pull request by leandromoreira**
Points of improvements:
* find a small docker image to be used as its base.
* push this image to docker hub so people can use it remotely without even cloning this repo.
* fix the python warning https://github.com/matplotlib/matplotlib/issues/5836 .

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM        ubuntu
+RUN apt-get update -qq && apt-get install -y \
+      build-essential git gcc python-numpy python-scipy python-matplotlib \
+      ipython ipython-notebook python-pandas python-sympy python-nose \
+      python2.7 python-setuptools python-dev
+RUN easy_install pip
+RUN pip install --upgrade scikit-learn
+RUN pip install --upgrade h5py
+RUN git clone --depth 1 https://github.com/Netflix/vmaf.git wmaf
```
Comment (li-zhi) [inline Dockerfile]: Please change all 'wmaf's to 'vmaf'.

Comment (li-zhi) [inline Dockerfile]: Or is 'wmaf' intentional?

File: README.md [hunk 2]
```diff
@@ -73,6 +73,41 @@ export PYTHONPATH=[path_to_repo_dir]/python:$PYTHONPATH
...
+
+# encoding it to vp9 (bitrate 700kpbs, vcodec VP9)
+docker run -v [src]:/files jrottenberg/ffmpeg -i /files/big_buck_bunny_360p_5mb.mp4 -c:v libvpx-vp9 -b:v 700K -c:a libvorbis /files/big_buck_bunny_360p.webm
+
+# extracting the yuv
+docker run -v [src]:/files jrottenberg/ffmpeg -i /files/big_buck_bunny_360p_5mb.mp4 -c:v rawvideo -pix_fmt yuv420p /files/360p_mpeg4-v_1000.yuv
+docker run -v [src]:/files jrottenberg/ffmpeg -i /files/big_buck_bunny_360p.webm -c:v rawvideo -pix_fmt yuv420p /files/360p_vp9_700.yuv
+
+# checking the WMAF score
+docker run -v [src]:/files vmaf yuv420p 640 368 /files/360p_mpeg4-v_1000.yuv /files/360p_vp9_700.yuv json
```
Comment (li-zhi) [inline README.md]: Should it be:
docker run -v [src]:/files vmaf yuv420p 640 368 /files/360p_mpeg4-v_1000.yuv /files/360p_vp9_700.yuv --out-fmt json
?

File: README.md [hunk 3]
```diff
@@ -73,6 +73,41 @@ export PYTHONPATH=[path_to_repo_dir]/python:$PYTHONPATH
 
 You can also add it to environment permanently. On Ubuntu, append the line above to *~/.bashrc* and run `source ~/.bashrc`. On Mac OS X, append it to *~/.bash_profile* and run `source ~/.bash_profile`.
 
+##Usage through Docker
+
+After cloning the WMAF repo, cd to the repo directory and run:
+
+```
+docker build -t vmaf .
```
Comment (li-zhi) [inline README.md]: Excuse my unfamiliarity with docker. When I run this command under mac, i got the following message:
unable to prepare context: Cannot locate Dockerfile: absDockerfile: "/Users/zli/Temp/test_vmaf/leandromoreira_vmaf/Dockerfile"

File: README.md (hunk 2 above)
Comment (leandromoreira) [inline README.md]: Yes, thank you ;) My bad again.

**Comment by leandromoreira** [comment]
Thanks @li-zhi for the suggestions, I've done them.

**Comment by leandromoreira** [comment]
Just in case, anyone is interested, the file `big_buck_bunny_360p.webm` (bitrate: 600kbps, vcodec: VP9, size: 3.5M) got `94.89` VMAF in comparision with its pivot, the file `big_buck_bunny_360p_5mb.mp4` (bitrate: 1000kbps, vcodec: MPEG-4 Visual, size: 5M), pretty good and it still saves 1.5M.

```json
    "aggregate": {
        "VMAF_feature_adm2_score": 0.9881909590019371,
        "VMAF_feature_motion_score": 2.6486005151515153,
        "VMAF_feature_vif_scale0_score": 0.86592472335017001,
        "VMAF_feature_vif_scale1_score": 0.97696951571782942,
        "VMAF_feature_vif_scale2_score": 0.98862492295492921,
        "VMAF_feature_vif_scale3_score": 0.99361265745622163,
        "VMAF_score": 94.891703917397805,
        "method": "mean"
    }
```

**Comment by li-zhi** [comment]
@leandromoreira Looks good. I am merging the change to master. Thanks for the contribution!

**Comment by leandromoreira** [comment]
Cool, thank you! @li-zhi do you have any plans to publish this image at [docker-hub](https://hub.docker.com/) so people can use it without even cloning this repo? (like I did with ffmpeg) Otherwise, I can push a docker image to docker-hub myself.

If you push the image to docker hub the readme needs to be updated (we can get rid of the `build` and `run` part):

```
docker run --rm -v [src]:/files netflix/vmaf yuv420p 640 368 /files/360p_mpeg4-v_1000.yuv /files/360p_vp9_700.yuv --out-fmt json

```

**Comment by li-zhi** [comment]
@leandromoreira I am thinking of adding docker support for other CLIs besides run_vmaf, such as run_vmaf_training, run_testing, and wrapper/vmafossexec, before pushing an image to docker-hub. What do you think?

**Comment by leandromoreira** [comment]
@li-zhi I think this is the ideal :) (since people will want to use the other CLI's as well), good catch.

**Comment by leandromoreira** [comment]
One possible solution is:

```diff
-ENTRYPOINT  ["/vmaf/run_vmaf"]
+WORKDIR ["/vmaf/"]
```

And then, I think this should work:

```bash
docker run --rm -v [src]:/files run_vmaf 
docker run --rm -v [src]:/files run_vmaf_training 
```

I also think we should tag the docker image and we should force a commit (or tag) on Dockerfile `git clone` operation, therefore a new version of vmaf will also mean a new image (clients can use an old image and only migrate when they feel comfortable)

**Comment by leandromoreira** [comment]
http://blog.kunicki.org/blog/2016/02/12/multiple-entrypoints-in-docker/

**Comment by li-zhi** [comment]
Awesome! Please go ahead and send a pull request.

**Pull request merged**